core/testdata/testspecs/v2_specs.go

```diff
@@ -864,26 +869,72 @@ ds -> ds_parse -> ds_multiply;
 	return StreamSpec{StreamSpecParams: params, toml: toml}
 }
 
-type WorkflowSpec struct {
+// WorkflowJobSpec is a test helper that wraps both the TOML and job.Job representation of a workflow job spec
+type WorkflowJobSpec struct {
 	toml string
+	j    job.Job
 }
 
-func (w WorkflowSpec) Toml() string {
+func (w WorkflowJobSpec) Toml() string {
 	return w.toml
 }
 
-func GenerateWorkflowSpec(id, owner, name, spec string) WorkflowSpec {
+func (w WorkflowJobSpec) Job() job.Job {
+	return w.j
+}
+
+// GenerateWorkflowJobSpec creates a WorkflowJobSpec from the given workflow yaml spec string
+func GenerateWorkflowJobSpec(t *testing.T, spec string) WorkflowJobSpec {
+	t.Helper()
+	s, err := pkgworkflows.ParseWorkflowSpecYaml(spec)
+	require.NoError(t, err, "failed to parse YAML workflow spec %s", spec)
+	id := s.CID
 	template := `
 type = "workflow"
 schemaVersion = 1
 name = "test-spec"
 workflowId = "%s"
-workflowOwner = "%s"
-workflowName = "%s"
 workflow = """
 %s
 """
 `
-	toml := fmt.Sprintf(template, id, owner, name, spec)
-	return WorkflowSpec{toml: toml}
-}
+
+	toml := fmt.Sprintf(template, id, spec)
+	j, err := workflows.ValidatedWorkflowJobSpec(toml)
+	require.NoError(t, err, "failed to validate TOML job spec for workflow %s", toml)
+	return WorkflowJobSpec{toml: toml, j: j}
+}
+
+func DefaultWorkflowJobSpec(t *testing.T) WorkflowJobSpec {
+	return GenerateWorkflowJobSpec(t, defaultWFYamlSpec)
+}
+
+var defaultWFYamlSpec = `
```

Comment:
Could this be pulled from common?

```suggestion
var defaultWFYamlSpec = WFYamlSpec(t, "myworkflow", "0x00000000000000000000000000000000000000aa")
```